includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -64,10 +64,6 @@ class AMP_Core_Theme_Sanitizer extends AMP_Base_Sanitizer {
 					'twentytwenty_skip_link_focus_fix', // See <https://github.com/WordPress/twentynineteen/pull/47>.
 				],
 			],
-			'add_smooth_scrolling'                    => [
```

Comment:
It's annoying that we need to move this piece of logic just because it needs to be done conditionally now.

In the past I've used closures for something like this. So, for a generic configuration system like we have here, the property getter would check if the property is callable first, and execute it (with some basic context) to fetch the actual dynamic property if that's the case.

So, this would stay in place but turn into something like this instead:
```php
'add_smooth_scrolling' => function ( $some_context ) {
	return ( 'twentytwenty' === get_template() && version_compare( get_bloginfo( 'version' ), '5.3.1', '<' ) )
		? [ '//a[ starts-with( @href, "#" ) and not( @href = "#" )and not( @href = "#0" ) and not( contains( @class, "do-not-scroll" ) ) and not( contains( @class, "skip-link" ) ) ]' ]
		: []
	}
},
```
Not sure we want something like there but I wanted to mention that possibility nevertheless.

Reply:
Agreed. Using a convention such as that outlined above would be one way of preventing further logic from being moved elsewhere to then be applied conditionally.

Suggestions @westonruter?

Reply:
Yes, that sounds good. It could be simply a matter of refactoring `\AMP_Core_Theme_Sanitizer::$theme_features` to instead be a method like `\AMP_Core_Theme_Sanitizer::get_theme_features()`

Reply:
Done in https://github.com/ampproject/amp-wp/pull/3912/commits/961cd23af1b7673abddaecf8b740acf5ec4a96eb.